forte/data/readers/openie_reader.py

```diff
@@ -0,0 +1,131 @@
+        To use this Reader, you must follow the dataset format described
+        `here
+        <https://github.com/gabrielStanovsky/oie-benchmark/tree/master/oie_corpus>`__.
+    """
+
+    def _collect(self, *args, **kwargs) -> Iterator[Any]:
+        # pylint: disable = unused-argument
+        r"""Should be called with param ``oie_directory`` which is a path to a
+        folder containing json files.
+
+        Args:
+            args: args[0] is the directory to the open ie files.
+            kwargs:
+
+        Returns: Iterator over files in the path with oie extensions.
+        """
+        oie_directory = args[0]
+        oie_file_extension = self.configs.oie_file_extension
+        logging.info("Reading dataset from %s with extension %s",
+                     oie_directory, oie_file_extension)
+        return dataset_path_iterator(oie_directory, oie_file_extension)
+
+    def _cache_key_function(self, oie_file: str) -> str:
+        return os.path.basename(oie_file)
+
+    def _parse_pack(self, file_path: str) -> Iterator[DataPack]:
+        pack = self.new_pack()
+        with open(file_path, "r", encoding="utf8") as f:
+            doc = f.readlines()
+
+        text = ""
+        offset = 0
+
+        for line in doc:
+            line = line.strip()
+            if line != "":
+                oie_component = line.split("\t")
+                sentence = oie_component[0]
+
+                # Add sentence.
+                Sentence(pack, offset, offset + len(sentence))
+                offset += len(sentence) + 1
+                text += sentence + " "
+
+                predicate = oie_component[1]
+
+                # Add predicate.
+                predicate_mention = PredicateMention(pack,
+                                                     offset,
+                                                     offset + len(predicate))
+                offset += len(predicate) + 1
+                text += predicate + " "
+
+                for arg in oie_component[2:]:
+                    # Add predicate argument.
+                    predicate_arg = PredicateArgument(pack,
+                                                      offset,
+                                                      offset + len(arg))
+                    offset += len(arg) + 1
+                    text += arg + " "
+
+                    # Add predicate link.
+                    PredicateLink(pack, predicate_mention, predicate_arg)
+
+        pack.set_text(text, replace_func=self.text_replace_operation)
+
+        Document(pack, 0, len(text))
+
+        pack.pack_name = file_path
+
+        yield pack
+
+    @classmethod
+    def default_configs(cls):
+        config = super().default_configs()
+
+        # Add OIE dataset file extension. The default is '.oie'
+        config.update({
+            'oie_file_extension': 'oie'
+        })
+        return config
+
+    def initialize(self, resources: Resources, configs: Config):
```

Comment:
Nice. Just my personal preferences, I like to put the `initialize` function just after `__init__`

Reply:
But the `__init__` is not here in this class right? Should I put this as the first method in this class?

Reply:
Oh yeah, I mean to place the initialization code earlier in the class. This is not a very strict requirement anyway.

Reply:
Done

Reply:
Learnt a lot from you code. Thanks!